src/flamegraph/flamegraph.js

```diff
@@ -72,9 +72,9 @@ window.addEventListener("click", function(e) {
 
         // set parameters for zoom state
         var el = target.querySelector("rect");
-        if (el && el.attributes && el.attributes.y && el.attributes._orig_x) {
+        if (el && el.attributes && el.attributes.y && el.attributes["fg:x"]) {
             var params = get_params()
-            params.x = el.attributes._orig_x.value;
+            params.x = el.attributes["fg:x"].value;
```

Comment:
Given that this gets saved into the URL (with `replaceState` below), I wonder if we also need to update the code that parses the URL to automatically restore the zoom appropriately (based on sample count rather than x position).

Reply:
I've come up with a commit that fixes the selector. I'm really quite new to JS and all this, and I don't understand why `fg\\:x` and `fg|x` don't work, but `*|x` does. I suppose what I came up with is potentially not the semantics we want, but I can't figure out anything else that works.

Reply:
Oh, that's bizarre. Doing some digging, I found [6.3.3 Attribute selectors and namespaces in the CSS spec](https://drafts.csswg.org/selectors-3/#attrnmsp), which states that:
> An attribute selector with an attribute name containing a namespace prefix that has not been previously declared is an invalid selector. 

That makes me think that you can't query for `fg|x` since `fg` hasn't been declared in the CSS "world", only in the HTML "world" (through the namespace tag). And with some more digging, it seems like [there isn't a way to specify a CSS namespace from JS](https://stackoverflow.com/a/23047888/472927). So, your solution is the only one it appears — just match independent of the namespace. Weird stuff.

If you've tested this and it works, then I'm happy with it! Should just be a matter of zooming through a bunch of different elements, and then using your browser back and forward arrows and seeing that the zoom changes as expected.

Reply:
> If you've tested this and it works, then I'm happy with it! Should just be a matter of zooming through a bunch of different elements, and then using your browser back and forward arrows and seeing that the zoom changes as expected.

I'm not sure forward and back have ever worked. As far as I can tell, they do not currently work on the latest-released version. What I broke then fixed with the latest commit was the ability to copy a zoomed URI to a new browser window and get the correct zoomed-in view upon pasting it.

Reply:
Ah, I see — I think that's because we're using `replaceState` rather than `pushState`, but that's not something you have to fix here given that it's already broken!